src/preamble.js

```diff
@@ -1105,6 +1105,10 @@ function createWasm() {
             err('falling back to ArrayBuffer instantiation');
             return instantiateArrayBuffer(receiveInstantiatedSource);
           });
+      }).catch(function(error) {
+        err("could not download wasm file");
+        err("falling back to ArrayBuffer instantiation");
+        return instantiateArrayBuffer(receiveInstantiatedSource);
```

Comment:
doesn't this code path also try to download the wasm? in what situations can it succeed when the fetch failed?

Reply:
This code path is just the catch statement for the failed fetch of the wasm file. There is catch above on the `WebAssembly.instantiateStreaming` promise, but not on the fetching of the wasm file itself. Just as in the case where the wasm file fails to instantiate, if the file itself is failed to be fetched, we will default to the ArrayBuffer instanciation.

Reply:
Sorry, I missed this PR in my email.

I still don't follow this. Let's say fetching the wasm fails, how can we succeed to do the ArrayBuffer instantiation? That will also do a network fetch. So if there is a network problem, this won't help us that I can see?